### PR TITLE
updated cosmic ray masking code to match latest from CTC software

### PR DESCRIPTION
**Pull request by kjl0025**
There was a software update after Ell had made the port.  This is the updated version.

### CODE REVIEW

**Comment by kjl0025** [comment]
Tagging @mygouf so she is aware of this (for the June deadline).

File: tests/test_cr_detection.py [hunk 1]
```diff
@@ -22,21 +22,21 @@ def find_plateaus_iit(streak_row, fwc, sat_thresh, plat_thresh, cosm_filter):
...
-            Full well capacity of detector *in DNs*.  Note that this may require a
-            conversion as FWCs are usually specified in electrons, but the image
-            is in DNs at this point.
-        sat_thresh (float): 
-            Multiplication factor for fwc that determines saturated cosmic pixels.
-        plat_thresh (float): 
-            Multiplication factor for fwc that determines edges of cosmic plateu.
-        cosm_filter (int): 
-            Minimum length in pixels of cosmic plateus to be identified.
+        Row with possible cosmics.
```
Comment (semaphoreP) [inline tests/test_cr_detection.py:25]: this lack of indentation makes it a bit difficult to read. can we indent it like the other docstrings?

File: tests/test_cr_detection.py [hunk 2]
```diff
@@ -22,21 +22,21 @@ def find_plateaus_iit(streak_row, fwc, sat_thresh, plat_thresh, cosm_filter):
...
+        Multiplication factor for fwc that determines saturated cosmic pixels.
+        plat_thresh (float):
+        Multiplication factor for fwc that determines edges of cosmic plateu.
+        cosm_filter (int):
+        Minimum length in pixels of cosmic plateus to be identified.
 
     Returns:
-        i_beg (array_like, int): 
-            Index of plateau beginnings, or None if there is no plateau.
+        i_beg (array_like, int):
```
Comment (semaphoreP) [inline tests/test_cr_detection.py:38]: same comment on how Returns docstrings should be formatted.

File: tests/test_cr_detection.py [hunk 3]
```diff
@@ -70,34 +77,51 @@ def remove_cosmics_iit(image, fwc, sat_thresh, plat_thresh, cosm_filter):
...
-            Full well capacity of detector *in DNs*.  Note that this may require a
-            conversion as FWCs are usually specified in electrons, but the image
-            is in DNs at this point.
-        sat_thresh (float): 
-            Multiplication factor for fwc that determines saturated cosmic pixels.
-        plat_thresh (float): 
-            Multiplication factor for fwc that determines edges of cosmic plateu.
-        cosm_filter (int): 
-            Minimum length in pixels of cosmic plateus to be identified.
+        image (array_like, float):
```
Comment (semaphoreP) [inline tests/test_cr_detection.py:80]: indent the text here too please.

Comment (semaphoreP) [inline tests/test_cr_detection.py:80]: indent the text here too please.

File: tests/test_cr_detection.py (hunk 2 above)
Comment (semaphoreP) [inline tests/test_cr_detection.py:38]: looks like this wasn't caught last time regarding the syntax of the Returns

File: tests/test_cr_detection.py [hunk 4]
```diff
@@ -70,34 +77,51 @@ def remove_cosmics_iit(image, fwc, sat_thresh, plat_thresh, cosm_filter):
...
+        tail length is longer than the length to the end of the image-area row,
+        the mask is truncated at the end of the row.
+        If 'full', a full-frame input is assumed, and if the input tail length
+        is longer than the length to the end of the full-frame row, the masking
+        continues onto the next row.  Defaults to 'image'.
 
     Returns:
-        mask (array_like, int): 
-            Mask for pixels that have been set to zero.
+        mask (array_like, int):
```
Comment (semaphoreP) [inline tests/test_cr_detection.py:111]: looks like this wasn't caught in the past either.

**Review by semaphoreP — APPROVED** [review]
just a few docstring comments. looks good! Will work on the bigger PR about noise maps next.

**Comment by kjl0025** [comment]
Thanks!  And all the comments should be addressed with the last push.

**Review by semaphoreP — APPROVED** [review]

**Pull request merged**